src/spec-configuration/controlManifest.ts

```diff
@@ -0,0 +1,92 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------------------------------------------*/
+
+import { promises as fs } from 'fs';
+import * as path from 'path';
+import * as os from 'os';
+import * as jsonc from 'jsonc-parser';
+
+import { request } from '../spec-utils/httpRequest';
+import * as crypto from 'crypto';
+import { Log } from '../spec-utils/log';
+
+export interface DisallowedFeature {
+	featureIdPrefix: string;
+	documentationURL?: string;
+}
+
+export interface DevContainerControlManifest {
+	disallowedFeatures: DisallowedFeature[];
+}
+
+const controlManifestPath = path.join(os.homedir(), '.devcontainer', 'cache', 'control-manifest.json');
```

Comment:
I think it may be nice to make this configurable. For example, we use the `--user-data-folder` flag in Codespaces to write to our "persisted folder" (persisted across starts/stops of codespaces outside of the container).